### DISCUSSION TITLE
Want one?

### DISCUSSION THREAD

**Discussion by mebs** [mebs/CM4-NAS#9] [General]
You can already order the PCB from manufacturers like JLC PCB just by uploading the Fab.zip files in this repo (make sure to take the latest version).
However you will have to buy all the components separately and solder them yourself.
I should also add that I am by no means a trained professional. This board is just a cool experiment that worked out well. I highly encourage you to get your hands dirty but just keep in mind that this PCB was never validated by someone who actually knows what they're doing 😄 

The next step for this project would be to look into pick-and-place assembly that JLC offers. For this to work, I would have to update the design to make sure all components are available for the assembly process and then generate special files.
This would be very cool as you could order fully assembled PCBs without having to open the KiCad project once.

[EDIT: PCB assembly with JLC PCB seems impossible for now as they don't offer the mezzanine connector for the CM4 at the moment, due to shortages, and we can't really get around that...]

If you have experience with ordering pick-and-place from JLC feel free to contribute to the project!
If you would be interested in such a thing leave a like! I kind of closed this book after assembling my first working NAS, but if enough people ask for it and no one contributed I'll look into it myself.

**Comment by kelnos** [comment]
Just added a thumbs-up to your post to show my interest... I figure that will reduce "me too" spam for anyone subscribed. But wanted to also make a note here to call attention to it, and for future possible-commenters :)

**Reply by mebs** [reply]
That goes to show how little I expect! 
Good thinking though :)

**Comment by konkav-js** [comment]
Hi. I just read about this on HaD, and found this interesting. So first thing went to easyeda, to find out whether they have those connectors for pcba. And it's kind of interesting, I found it in the "Extended Part" list, but when I select it, it says that it isn't supported for assembly, but they are also out of stock. But could find other out of stock connectors. and they had no "doesn't supported" sign, at least some of out of stock connectors.

I think, I won't be interested in pcba, if I need to hand solder those connectors in. But if this is just a glitch, and it will be available...

And it is fascinating to see, that making a board with GBE and PCIe are so close to average hobbyist.

**Reply by mebs** [reply]
I assume you're talking about the high density connectors for the CM4. 
Yes they are often out of stock but they were restocked a couple times in the past months. Here's the part on LCSC:
https://lcsc.com/product-detail/Mezzanine-Connectors-Board-to-Board_HRS-Hirose-DF40C-100DS-0-4V-51_C597931.html

I just took a look at what mezzanine connectors are available for PCB assembly with JLC and this connector is not there anymore... There are other smaller versions of this connector but not this one. A few months back I saw it in the list, so maybe they'll put it back, but either way I think I'll contact them to know for sure if we can rely on this connector being available for assembly in the future.

**Reply by mebs** [reply]
> And it is fascinating to see, that making a board with GBE and PCIe are so close to average hobbyist.

Yes!! This is amazing. KiCad being free and open source, it has a great community which makes it easier to learn.

**Reply by konkav-js** [reply]
Did you get any answer from them?

It would be good, if they would put back those connectors for pcba. I mean, I have no hot air soldering station. And while it is easy to buy one, I have no practice using one. And I presume, a long, thin and narrow plastic connector is not the best thing to learn to solder with hot air 😁

**Reply by mebs** [reply]
I did not contact them yet, will probably do that in the weekend.
Soldering those connectors is not as bad as I thought, it's doable, but it's still not fun 😅

**Reply by konkav-js** [reply]
After searching around a little, found someone who hand soldered it to a cnc milled pcb ([link](https://hackaday.io/project/175832/gallery#9da6076c9b40558c18dd3e3d292e99f7)) I feel more confident now 😃 I think it's time to update my kicad from stable to nightly.

**Reply by mebs** [reply]
Oh yes it's totally doable, if you look at the Readme I shared this video:
https://www.youtube.com/watch?v=eukcrFc18P4

This is where I started and just following the tutorial gave me results. Later I found that the less solder paste you use the better.

EDIT: my bad, just realized you meant with a soldering iron and not with hot air. I would still recommend getting a cheap hot air gun but let us know if you manage! Good luck!
